pkg/elemental/pages/index.vue

```diff
@@ -1,594 +1,58 @@
-  ELEMENTAL_SCHEMA_IDS,
-  ELEMENTAL_CLUSTER_PROVIDER,
-  KIND
-} from '../config/elemental-types';
-import { createElementalRoute } from '../utils/custom-routing';
-import { filterForElementalClusters } from '../utils/elemental-utils';
-import BuildIso from '../components/BuildIso';
-
-const MAX_ITEMS_PER_TABLE = 6;
+import InstallView from '../components/InstallView';
+import DashboardView from '../components/DashboardView';
+import { ELEMENTAL_SCHEMA_IDS } from '../config/elemental-types';
 
 export default {
-  name:       'Dashboard',
+  name:       'BaseView',
   components: {
-    Loading, Banner, PercentageBar, ResourceTable, BuildIso
+    Loading,
+    InstallView,
+    DashboardView
   },
   async fetch() {
     // this covers scenario where Elemental Operator is deleted from Apps and we lose the Elemental Admin role for Standard Users...
     if (this.$store.getters['management/canList'](ELEMENTAL_SCHEMA_IDS.MACHINE_REGISTRATIONS)) {
-      const requests = {
-        // needed for table
-        machineRegistrations: this.$store.dispatch('management/findAll', { type: ELEMENTAL_SCHEMA_IDS.MACHINE_REGISTRATIONS }),
-        // needed for resource gauge
-        machineInventories:   this.$store.dispatch('management/findAll', { type: ELEMENTAL_SCHEMA_IDS.MACHINE_INVENTORIES }),
-        rancherClusters:      this.$store.dispatch('management/findAll', { type: CAPI.RANCHER_CLUSTER }),
-        // needed for table
-        managedOsImages:      this.$store.dispatch('management/findAll', { type: ELEMENTAL_SCHEMA_IDS.MANAGED_OS_IMAGES }),
-        // needed to populate cluster name col on machine inventories list
-        machineInvSelector:   this.$store.dispatch(`management/findAll`, { type: ELEMENTAL_SCHEMA_IDS.MACHINE_INV_SELECTOR }),
-        elementalSchema:      this.$store.getters['management/schemaFor'](ELEMENTAL_SCHEMA_IDS.MACHINE_INVENTORIES)
-      };
-
+      let installedApps;
       // needed to check if operator is installed
       if (this.$store.getters['management/canList'](CATALOG.APP)) {
-        requests.installedApps = this.$store.dispatch('management/findAll', { type: CATALOG.APP });
+        installedApps = await this.$store.dispatch('management/findAll', { type: CATALOG.APP });
       }
 
-      const allDispatches = await allHash(requests);
-
-      this.resourcesData = {};
-
-      this.resourcesData[ELEMENTAL_SCHEMA_IDS.MACHINE_REGISTRATIONS] = allDispatches.machineRegistrations;
-      this.resourcesData[ELEMENTAL_SCHEMA_IDS.MACHINE_INVENTORIES] = allDispatches.machineInventories;
-      this.resourcesData[this.ELEMENTAL_CLUSTERS] = filterForElementalClusters(allDispatches.rancherClusters);
-      this.resourcesData[ELEMENTAL_SCHEMA_IDS.MANAGED_OS_IMAGES] = allDispatches.managedOsImages;
-      this.resourcesData[ELEMENTAL_SCHEMA_IDS.MACHINE_INV_SELECTOR] = allDispatches.machineInvSelector;
+      const elementalSchema = this.$store.getters['management/schemaFor'](ELEMENTAL_SCHEMA_IDS.MACHINE_INVENTORIES)
```

Comment:
Will the schemas always be populated in the store at this point?

Reply:
yes they will, as soon as the operator has been installed